asv_bench/benchmarks/benchmarks.py

```diff
@@ -376,38 +376,63 @@ def time_head(self, shape, head_count):
         execute(self.df.head(self.head_count))
 
 
-class TimeFillna:
-    param_names = ["self_type", "value_type", "shape", "limit", "inplace"]
+class TimeFillnaSeries:
+    param_names = ["value_type", "shape", "limit"]
     params = [
-        ["DataFrame", "Series"],
-        ["scalar", "dict", "DataFrame", "Series"],
+        ["scalar", "dict", "Series"],
         UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE],
         [None, 0.8],
-        [False, True],
     ]
 
-    def setup(self, self_type, value_type, shape, limit, inplace):
+    def setup(self, value_type, shape, limit):
         pd = IMPL[ASV_USE_IMPL]
-        columns = [f"col{x}" for x in range(shape[1])]
-        if self_type == "DataFrame":
-            self.dataset = pd.DataFrame(
-                np.nan, index=pd.RangeIndex(shape[0]), columns=columns
+        self.dataset = pd.Series(np.nan, index=pd.RangeIndex(shape[0]))
+
+        if value_type == "scalar":
+            self.value = 18.19
+        elif value_type == "dict":
+            self.value = {k: k * 1.23 for k in range(shape[0])}
+        elif value_type == "Series":
+            self.value = pd.Series(
+                [k * 1.23 for k in range(shape[0])], index=pd.RangeIndex(shape[0])
             )
-        elif self_type == "Series":
-            self.dataset = pd.Series(np.nan, index=pd.RangeIndex(shape[0]))
         else:
             assert False
+        limit = int(limit * shape[0]) if limit else None
+        self.kw = {"value": self.value, "limit": limit}
+
+    def time_fillna(self, value_type, shape, limit):
+        execute(self.dataset.fillna(**self.kw))
+
+    def time_fillna_inplace(self, value_type, shape, limit):
+        self.dataset.fillna(inplace=True, **self.kw)
+        execute(self.dataset)
+
+
+class TimeFillnaDataFrame:
+    param_names = ["value_type", "shape", "limit", "inplace"]
+    params = [
+        ["scalar", "dict", "DataFrame", "Series"],
+        UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE],
+        [None, 0.8],
+    ]
+
+    def setup(self, value_type, shape, limit):
+        pd = IMPL[ASV_USE_IMPL]
+        columns = [f"col{x}" for x in range(shape[1])]
+        self.dataset = pd.DataFrame(
+            np.nan, index=pd.RangeIndex(shape[0]), columns=columns
+        )
```

Comment:
what about moving `nan` data generation to a separate cache-able generator to `utils.py`, as we did for other generators?
https://github.com/modin-project/modin/blob/0bc409d6ae294986db5f7db19d082622544fb122/asv_bench/benchmarks/utils.py#L137

This will reduce the benchmarking time (because of the caching) and will allow us to easily add parametrization for the density of the `nan` data in the future

Reply:
Good point!
Data generation is moved to separate function.